### DISCUSSION TITLE
플러그인과 컴포넌트의 차이?

### DISCUSSION THREAD

**Discussion by menthamin** [Well-Founded-Confidence/RealMySQL#27] [Q&A]
책에서는 (89~90p) 컴포넌트는 상호작용이 가능하다고 하는데 둘의 차이를 명확히 모르겠습니다.

**Comment by menthamin** [comment]
플러그인 == 모듈 == 컴포넌트 라고 생각을..했지만
플러그인 != 모듈 != 컴포넌트 인 것 같다.

플러그인에서 컴포넌트로 올라가는 경우가 있다.

플러그인은 직접 하드코딩을 해서 MySQL에 적용해야 한다.
컴포넌트는 MySQL에 보다 안전하고 쉽게 적용할 수 있다.

**플러그인 간에 상호 의존성 문제를 컴포넌트로 해결**할 수 있다.

**Comment by nahyunLee** [comment]
1. Plugin: In MySQL, a plugin is a dynamically loaded library that adds new functionality or extends the existing functionality of the MySQL server. Plugins can be used to add new storage engines, authentication mechanisms, and other features to the MySQL server. Plugins can be loaded and unloaded at runtime, without requiring a server restart. Examples of MySQL plugins include the InnoDB storage engine, the authentication plugins, and the audit plugins.

2. Component: A component in MySQL refers to a specific type of plugin that is used to add functionality to the MySQL client programs, such as the mysql command-line client or the MySQL Workbench GUI tool. MySQL components are also dynamically loaded and unloaded at runtime. Examples of MySQL components include the authentication components, the encryption components, and the scripting components.

In summary, a MySQL plugin is a dynamically loaded library that adds functionality to the MySQL server, while a MySQL component is a plugin that adds functionality to the MySQL client programs.


역시나 chatGPT의 답변입니다...ㅎ

**Reply by CODe5753** [reply]
서버에 기능추가, 클라이언트에 기능 추가.. 인가요?
